src/client.cc

```
@@ -92,6 +93,114 @@ struct AlignedBuffer {
+// The driver API copies operate on the calling thread's *current* context.
+// GetObjectAsync/PutObjectAsync dispatch through std::async, so a fallback can
+// run on a pool thread that has never touched CUDA; there every copy would fail
+// with CUDA_ERROR_INVALID_CONTEXT. Borrow the primary context of the device the
+// buffer belongs to for the duration of the copy, and put the thread back the
+// way we found it afterwards. A thread that already has a context is left
+// alone.
+class ScopedCudaContext {
+ public:
+  ScopedCudaContext(const CudaHostCopy& fns, void* devptr) : fns_(&fns) {
+    void* current = nullptr;
+    if (fns.ctx_get(&current) != 0) return;
```

Cppcheck (2.21.0)

[style] The function 'JsonTypeToString' is never used.

(unusedFunction)